java/src/main/java/org/digma/intellij/plugin/idea/psi/java/GrpcFramework.java

```diff
@@ -0,0 +1,133 @@
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.psi.search.searches.ClassInheritorsSearch;
+import com.intellij.util.Query;
+import org.digma.intellij.plugin.log.Log;
+import org.digma.intellij.plugin.model.discovery.DocumentInfo;
+import org.digma.intellij.plugin.model.discovery.EndpointInfo;
+import org.digma.intellij.plugin.model.discovery.MethodInfo;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+
+public class GrpcFramework {
+    private static final Logger LOGGER = Logger.getInstance(GrpcFramework.class);
+
+    public static final String BINDABLE_SERVICE_ANNOTATION_STR = "io.grpc.BindableService";
+    public static final String DIGMA_UNKNOWN_SERVICE_NAME = "Digma.Unknown.Grpc.Service";
+
+    private final Project project;
+
+    // late init
+    private boolean lateInitAlready = false;
+    private PsiClass bindableServiceAnnotationClass;
+
+    public GrpcFramework(Project project) {
+        this.project = project;
+    }
+
+    private void lateInit() {
+        if (lateInitAlready) return;
+
+        JavaPsiFacade psiFacade = JavaPsiFacade.getInstance(project);
+        bindableServiceAnnotationClass = psiFacade.findClass(BINDABLE_SERVICE_ANNOTATION_STR, GlobalSearchScope.allScope(project));
+
+        lateInitAlready = true;
+    }
+
+    private boolean isGrpcServerRelevant() {
+        return bindableServiceAnnotationClass != null;
+    }
+
+    public void endpointDiscovery(@NotNull PsiFile psiFile, @NotNull DocumentInfo documentInfo) {
+        lateInit();
+        if (!isGrpcServerRelevant()) {
+            return;
+        }
+
+        Query<PsiClass> grpcServerClassesInFile = ClassInheritorsSearch.search(bindableServiceAnnotationClass, GlobalSearchScope.fileScope(psiFile), true);
+
+        for (PsiClass currGrpcServerClass : grpcServerClassesInFile) {
+            if (JavaPsiUtils.isBaseClass(currGrpcServerClass)) {
+                // if has no super class then it is the generated GRPC server class, we do not want it
+                Log.log(LOGGER::debug, "endpointDiscovery, skip bindableService GrpcServerClass fqn='{}' since it is the generated GRPC base service", currGrpcServerClass.getQualifiedName());
+                continue;
+            }
+            Log.log(LOGGER::info, "endpointDiscovery, bingo - its a GRPC server class, its fqn='{}'", currGrpcServerClass.getQualifiedName());
+            addEndpointMethods(currGrpcServerClass, documentInfo);
+        }
+    }
+
+    protected void addEndpointMethods(@NotNull PsiClass grpcServerClass, @NotNull DocumentInfo documentInfo) {
+
+        String grpcServiceName = evaluateServiceName(grpcServerClass);
+        Log.log(LOGGER::info, "addEndpointMethods for grpcServerClass fqn='{}' with evaluated serviceName='{}'", grpcServerClass.getQualifiedName(), grpcServiceName);
```

Comment:
Do we really need here LOGGER::info? maybe we can change it to debug level?

Reply:
I will reduce it to debug